resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/config.scala

```diff
@@ -288,4 +288,37 @@ package object config {
       .doc("Interval between reports of the current app status in cluster mode.")
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefaultString("1s")
+
+  // Spark dependency server for submission v2
+
+  private[spark] val DEPENDENCY_SERVER_PORT =
+    ConfigBuilder("spark.kubernetes.dependencyserver.port")
+      .doc("Port for the Kubernetes dependency server to listen on.")
+      .intConf
+      .createWithDefault(10000)
+
+  private[spark] val DEPENDENCY_SERVER_KEY_PEM =
+    ConfigBuilder("spark.ssl.kubernetes.dependencyserver.keyPem")
+      .doc("Key PEM file to use when having the Kubernetes dependency server listen on TLS.")
+      .stringConf
+      .createOptional
+
+  private[spark] val DEPENDENCY_SERVER_CERT_PEM =
```

Comment:
I was planning to save all documentation for a final sweep where we eliminate all of the v1 submission logic and replace it with v2. At that point the docs should be updated to remove the old options and replace them with the new ones.